pool.h

```diff
@@ -0,0 +1,44 @@
+/**
+ *  Copyright 2014-2015 MongoDB, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+#ifndef __MONGODB_DRIVER_POOL_H__
+#define __MONGODB_DRIVER_POOL_H__
+
+extern "C" {
+#include "../../../libmongoc/src/mongoc/mongoc.h"
+}
+
+namespace HPHP {
+
+class Pool
+{
+	public:
+		mongoc_client_pool_t *m_pool;
+
+		Pool(const mongoc_uri_t *uri)
+		{
+			m_pool = mongoc_client_pool_new(uri);
```

Comment:
Does this mean that [connection pool options](https://docs.mongodb.org/manual/reference/connection-string/#connection-pool-options) will now be handled? Since the pool is only be created to preserve topology state between requests, I don't think we want it treated like a typical connection pool and allow multiple connections to be open.